frontend/src/component/FeedItem/Preview.jsx

```diff
@@ -164,16 +165,19 @@ class Preview extends React.Component {
   }
 
   componentDidUpdate (prevProps) {
-    if (prevProps.content === this.props.content) {
+    const { props } = this
+    if (prevProps.content === props.content) {
       this.testPreviewOverflow()
-    } else if (prevProps.content.current_revision_id !== this.props.content.current_revision_id) {
+    } else if (prevProps.content.commentList !== props.content.commentList ||
+      prevProps.content.currentRevisionId !== props.content.currentRevisionId) {
       this.updatePreview()
     }
   }
 
   shouldComponentUpdate (nextProps, nextState) {
     return (
-      nextProps.content.current_revision_id !== this.props.content.current_revision_id ||
+      nextProps.content.commentList !== this.props.content.commentList ||
+      nextProps.content.currentRevisionId !== this.props.content.currentRevisionId ||
```

Comment:
Maybe make put condition into a function since it becomes big and is also used in componentDidUpdate

Reply:
not exactly the same thing that componentDidUpdate, here we check nextProps, at componentDidUpdate we check prevProps, so the function will be something like

```jsx
foo = (thing1, content) => return (
      thing1.content.commentList !== content.commentList ||
      thing1.content.currentRevisionId !== content.currentRevision
)
```
and I will love some tip names

Reply:
not exactly the same thing that componentDidUpdate, here we check nextProps, at componentDidUpdate we check prevProps, so the function will be something like

```jsx
foo = (thing1, content) => return (
      thing1.content.commentList !== content.commentList ||
      thing1.content.currentRevisionId !== content.currentRevision
)
```
and I will love some tip names

Reply:
Maybe this?

```js
contentHasChanged = (oldContent, newContent) => return (
    oldContent.commentList !== newContent.commentList ||
    oldContent.currentRevisionId !== newContent.currentRevision
)
```

By the way, you are comparing lists directly, is it okay?

Reply:
In one case is `hasChanged` and in the other is `willChange`...

> By the way, you are comparing lists directly, is it okay?
Yes